internal/task/memory/task.go

```diff
@@ -79,14 +83,18 @@ func (r *DefaultTaskRegistrar) GetRegisteredTaskCount() uint {
 }
 
 func (r *DefaultTaskRegistrar) Processor(job interface{}) error {
+	logger.ApplicationLogger.Info("Received job of type", "type", fmt.Sprintf("%T", job))
 
 	// Perform a type switch to handle different job types
 	switch j := job.(type) {
 	case amqp.Delivery:
-		return r.amqpMsgProcessor(job)
+		return r.amqpMsgProcessor(j) // Pass the job as amqp.Delivery
+	case *schema.Signature:
```

Comment:
can we identify like this message is from redis as in future we can support redis cluster also.

Reply:
r.broker.BrokerType() ?

Reply:
we can @surendratiwari3  , I have added BrokerType function for that